pkgs/development/libraries/qmltermwidget/default.nix

```diff
@@ -38,6 +38,14 @@ stdenv.mkDerivation {
       hash = "sha256-1GjC2mdfP3NpePDWZaT8zvIq3vwWIZs+iQ9o01iQtD4=";
     })
 
+    # A fix to the above series of patches, to fix a crash in lomiri-terminal-app
+    # Remove (and update the above fetchpatch) when https://github.com/gber/qmltermwidget/pull/1 merged
+    (fetchpatch {
+      name = "0002-qmltermwidget-Mark-ColorSchemeManager-singleton-as-C++-owned.patch";
+      url = "https://github.com/gber/qmltermwidget/pull/1/commits/f3050bda066575eebdcff70fc1c3a393799e1d6d.patch";
```

Comment:
```suggestion
      url = "https://github.com/gber/qmltermwidget/commit/f3050bda066575eebdcff70fc1c3a393799e1d6d.patch";
```
i think is more stable tho i am unclear if these nodes ever get garbage collected once the change merges and a new hash is assigned.

Reply:
`https://github.com/gber/qmltermwidget/pull/1/commits/f3050bda066575eebdcff70fc1c3a393799e1d6d.patch` just redirects to `https://github.com/gber/qmltermwidget/commit/f3050bda066575eebdcff70fc1c3a393799e1d6d.patch`, but I like the extra bit of self-documentation that `pull/1/commits` in the patch URL gives.

If you think it's better to not use this indirection, then I think I would prefer:
```suggestion
      url = "https://github.com/OPNA2608/qmltermwidget/commit/f3050bda066575eebdcff70fc1c3a393799e1d6d.patch";
```

otherwise you get this upon manual inspection:
![image](https://github.com/NixOS/nixpkgs/assets/23431373/f36beb8c-24b2-42fc-a608-152096c7ec8f)

---

As for them getting garbage collected, I don't think they ever get removed. Even 5 year old merged PRs from deleted branches still work with this. It would likely break if GitHub does the nuclear ban option and purges all of my opened repos, issues & PRs, but I'm not sure what we *could* do in that case…

Reply:
i delete my forks regularly so i prefer the repo to match the origin but not fork -- i think it is easier to get reviewed too as the patch will match the repo even if it doesn't provide any additional assurances...

https://github.com/NixOS/nixpkgs/pull/268449#discussion_r1398348510 is the comment i got about the stability of `pulls/commits` url and i thought there was some best practices i read about using the `commit` one, but perhaps this is all urban legend.

Reply:
Tested that really quickly. Going via `pull/xyz/commits` *does* error out on force-pushes

- https://github.com/OPNA2608/nixpkgs/pull/11/commits/efc9f839adfd1c672c6d7dde714908cd7da5f3a2.patch

…but `commits` works fine

- https://github.com/OPNA2608/nixpkgs/commit/efc9f839adfd1c672c6d7dde714908cd7da5f3a2.patch

So that's definitely an interesting detail to keep in mind. Thanks!

As for the fork getting deleted, I don't plan to do that until the PR is merged & the `fetchpatch` here gets updated (+ prolly some more additional months, until I remember to clean up my forks) so I don't see this as an immediate issue.

Reply:
> As for the fork getting deleted, I don't plan to do that until the PR is merged & the `fetchpatch` here gets updated (+ prolly some more additional months, until I remember to clean up my forks) so I don't see this as an immediate issue.

though this code could live for a long time -- someone might really like the nixpkgs tarball that has your fork, so if you commit this change as is you can never delete the fork.

Reply:
ok -- i think that the commit url needs to point to the origin rather than a fork as it's possible for people to use older nixpkgs tarballs that would contain this patch and use older versions where this would be an issue if the fork is deleted -- regardless of if the patch was accepted upstream.

Reply:
> someone might really like the nixpkgs tarball that has your fork

In that case, the patch would still be available via (and should usually be automatically preferred to be fetched from) `https://cache.nixos.org`. That's what happens with the patch series that is already part of this package:
```
copying path '/nix/store/1abdcyn4ksd9saghszzxlprrac8yjm1c-0001-qmltermwidget-lomiri-submissions.patch' from 'https://cache.nixos.org'...
```

But I suppose if someone is doing a cache-less world rebuild… *Hmmm*, fine.

Reply:
Changed to `gber/qmltermwidget/commit`.